### PR TITLE
Decouple policy repository from Oracle rules fetching

### PR DESCRIPTION
**Pull request by Waldz**
Now it's possible to mock policy rules for running service wo rules actual fetching from Oracle

### CODE REVIEW

File: core/policy/oracle.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+		fetchURL:           policyURL,
+		fetchInterval:      interval,
+		fetchSubscriptions: make([]policySubscription, 0),
+		fetchShutdown:      make(chan struct{}),
+	}
+}
+
+// Start begins fetching policies to to subscribers
+func (pr *Oracle) Start() {
+	go pr.fetchLoop()
```
Comment (vkuznecovas) [inline core/policy/oracle.go]: this is redundant, rename fetchLoop to Start and allow the consumer to decide if it wants to run it synchronously or asynchronously.

File: core/policy/oracle.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	return &Oracle{
+		client:             client,
+		fetchURL:           policyURL,
+		fetchInterval:      interval,
+		fetchSubscriptions: make([]policySubscription, 0),
+		fetchShutdown:      make(chan struct{}),
+	}
+}
+
+// Start begins fetching policies to to subscribers
```
Comment (soffokl) [inline core/policy/oracle.go]:
```suggestion
// Start begins fetching policies to subscribers
```

File: core/policy/oracle.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+		index, exist := pr.getPolicyIndex(subscriptionsNew, policy)
+		if !exist {
+			index = len(subscriptionsNew)
+			subscriptionsNew = append(subscriptionsNew, policySubscription{
+				policy:      policy,
+				subscribers: []*Repository{repository},
+			})
+		}
+
+		var err error
```
Comment (soffokl) [inline core/policy/oracle.go]: You don't need to define var before if it used nowhere: 
```suggestion
		if err := pr.fetchPolicyRules(&subscriptionsNew[index]); err != nil {
```

File: core/policy/oracle.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+		case <-pr.fetchShutdown:
+			return
+		case <-time.After(pr.fetchInterval):
+			pr.fetchLock.Lock()
+
+			subscriptionsActive := make([]policySubscription, len(pr.fetchSubscriptions))
+			copy(subscriptionsActive, pr.fetchSubscriptions)
+
+			for index := range subscriptionsActive {
+				var err error
```
Comment (soffokl) [inline core/policy/oracle.go]: You don't need to define var before if it used nowhere: 
```suggestion
		if err := pr.fetchPolicyRules(&subscriptionsActive[index]); err != nil {
```

File: core/policy/oracle_test.go [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+			policy:      oracle.Policy("2"),
+			subscribers: []*Repository{repo},
+		},
+	)
+	return oracle
+}
+
+func mockPolicyServer() *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/1" {
```
Comment (soffokl) [inline core/policy/oracle_test.go]: > ifElseChain: rewrite if-else to switch statement (gocritic)

File: cmd/di_desktop.go [hunk 6]
```diff
@@ -238,15 +238,15 @@ func (di *Dependencies) bootstrapServiceComponents(nodeOptions node.Options, ser
 	}
 	di.ServiceSessionStorage = storage
 
-	di.PolicyRepository = policy.NewRepository(di.HTTPClient, servicesOptions.AccessPolicyAddress, servicesOptions.AccessPolicyFetchInterval)
-	di.PolicyRepository.Start()
+	di.PolicyOracle = policy.NewOracle(di.HTTPClient, servicesOptions.AccessPolicyAddress, servicesOptions.AccessPolicyFetchInterval)
+	di.PolicyOracle.Start()
```
Comment (vkuznecovas) [inline cmd/di_desktop.go]: is this not blocking now?

Comment (Waldz) [reply cmd/di_desktop.go]: my bad

**Pull request merged**